app/assets/javascripts/application.js

```diff
@@ -12,7 +12,11 @@
 //
 //= require jquery
 //= require jquery_ujs
-//= require react
-//= require react-server
 //= require react_ujs
-//= require ./components/test_component
\ No newline at end of file
+
+var React = window.ReactFoo = window.React = global.React = require('react');
```

Comment:
as proof of CommonJS being used (rather than any React assignment from like `react_ujs`) I've set `window.ReactFoo`